src/rational/ratSyntax.sml

```diff
@@ -17,10 +10,14 @@ val ERR = mk_HOL_ERR "ratSyntax";
  *--------------------------------------------------------------------------*)
 
 (*val int_ty = intSyntax.int_ty;*)
-val rat = mk_thy_type{Tyop = "rat", Thy="rat", Args = []};
+val rat_ty = mk_thy_type{Tyop = "rat", Thy="rat", Args = []};
 
+(* old definitions:
 val rat_0_tm = prim_mk_const {Name="rat_0",Thy="rat"};
 val rat_1_tm = prim_mk_const {Name="rat_1",Thy="rat"};
+ *)
+val rat_0_tm = “0q”;
```

Comment:
You shouldn't call the parser in library files without having ensured the parser’s behaviour at the head of the file.  For the sort of thing you need to do, see [this example](https://github.com/HOL-Theorem-Prover/HOL/blob/1d5cce12600fa6c2b79f88f35406216e78ef0d2b/src/taut/tautLib.sml#L21-L26). Otherwise, you run the risk that this code will be executed in a context where the global grammars make a mess of your attempt to parse any given string.